src/frontend/js/components/DashboardCreditCardsManagement/index.spec.tsx

```diff
@@ -0,0 +1,437 @@
+    })
+    .generate(),
+}));
+
+jest.mock('utils/indirection/window', () => ({
+  confirm: jest.fn(() => true),
+}));
+
+describe('<DashboardCreditCardsManagement/>', () => {
+  const createQueryClientWithUser = (isAuthenticated: Boolean) => {
+    const user = isAuthenticated ? mockFactories.UserFactory.generate() : null;
+    const { clientState } = mockFactories.PersistedClientFactory({
+      queries: [mockFactories.QueryStateFactory('user', { data: user })],
+    });
+    const client = createQueryClient();
+    hydrate(client, clientState);
+
+    return client;
+  };
+
+  beforeEach(() => {
+    fetchMock.get('https://joanie.endpoint/api/orders/', []);
+    fetchMock.get('https://joanie.endpoint/api/addresses/', []);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    fetchMock.restore();
+  });
+
+  it('renders an empty list with placeholder', async () => {
+    fetchMock.get('https://joanie.endpoint/api/credit-cards/', []);
+    const client = createQueryClientWithUser(true);
+    await act(async () => {
+      render(
+        <QueryClientProvider client={client}>
+          <IntlProvider locale="en">
+            <SessionProvider>
+              <DashboardTest initialRoute={DashboardPaths.PREFERENCES} />
+            </SessionProvider>
+          </IntlProvider>
+        </QueryClientProvider>,
+      );
+    });
+    // No error is shown.
+    expect(screen.queryByText('An error occurred', { exact: false })).toBeNull();
+    // The empty placeholder is shown.
+    screen.getByText("You haven't created any credit cards yet.");
+  });
+
+  it('renders the correct label for expired date', async () => {
+    const date = faker.date.past(0.2);
+    const creditCard: CreditCard = {
+      ...mockFactories.CreditCardFactory.generate(),
+      expiration_month: date.getMonth() + 1,
+      expiration_year: date.getFullYear(),
+    };
+    fetchMock.get('https://joanie.endpoint/api/credit-cards/', [creditCard]);
+    const client = createQueryClientWithUser(true);
+    await act(async () => {
+      render(
+        <QueryClientProvider client={client}>
+          <IntlProvider locale="en">
+            <SessionProvider>
+              <DashboardTest initialRoute={DashboardPaths.PREFERENCES} />
+            </SessionProvider>
+          </IntlProvider>
+        </QueryClientProvider>,
+      );
+    });
+    // No error is shown.
+    expect(screen.queryByText('An error occurred', { exact: false })).toBeNull();
+    const element = screen.getByText(
+      'Expired since ' +
+        (date.getMonth() + 1).toLocaleString(undefined, {
+          minimumIntegerDigits: 2,
+        }) +
+        '/' +
+        date.getFullYear(),
+    );
+    expect(element.className).toContain('dashboard-credit-card__expiration');
+    expect(element.className).toContain('dashboard-credit-card__expiration--expired');
+    expect(element.className).not.toContain('dashboard-credit-card__expiration--soon');
```

Comment:
I think use `classList` is more robust.
e.g: Currently the following test will pass
```js
// element = <div className="dashboard-credit-card__expiration--expired">...</div>
expect(element.className).toContain('dashboard-credit-card__expiration');
expect(element.className).toContain('dashboard-credit-card__expiration--expired');
```

```suggestion
    expect(element.classList.contains('dashboard-credit-card__expiration')).toBe(true);
    expect(element.classList.contains('dashboard-credit-card__expiration--expired')).toBe(true);
    expect(element.classList.contains('dashboard-credit-card__expiration--soon').toBe(false);
```